Don't highlight `$${}` as string interpolation

Ma27 opened the pull request:
Double dollar signs in Nix strings evaluate to a two dollar
signs rather than a dollar and a string interpolation:

```
Welcome to Nix version 2.1.3. Type :? for help.

nix-repl> ''
          $${foo}
          ''
"$${foo}\n"
```

Ma27 commented:
@Lnl7 it would be great if you could have a look at this! :)

LnL7 commented on syntax/nix.vim:
```diff
@@ -32,7 +32,7 @@ syn keyword nixTodo FIXME NOTE TODO OPTIMIZE XXX HACK contained
 syn region  nixComment start=+/\*+ end=+\*/+ contains=nixTodo,@Spell
 
-syn region nixInterpolation matchgroup=nixInterpolationDelimiter start="\${" end="}" contained contains=@nixExpr,nixInterpolationParam
+syn region nixInterpolation matchgroup=nixInterpolationDelimiter start="[^\$]\${" end="}" contained contains=@nixExpr,nixInterpolationParam
```
skip might also work for start/end, otherwise use `\zs` to exclude the preceding character from matching.

Ma27 commented:
> I didn't properly look at this yet because of the failing test.

Yeah, sorry, should've made more clear that I was unsure about the reason why the test failed.

I changed the PR to highlight `$$` to avoid confusion and added a testcase for it.

Ma27 commented:
@LnL7 anything else missing here? :)
